gateway/README.md

```diff
@@ -11,158 +11,263 @@ This basic example demonstrates how to expose a service or a function through an
+4. Fill the **Name** text box.
+5. Fill the **Hostname** text box. The name you entered is referred to as the **\{hostname\}**. The domain next to it is referred to as the **\{domain\}**.
+6. In the **Access strategies** section, click on the dropdown and select the **OAuth2** field. 
+7. Fill the **Required scope** text box with `read, write`.
+8. Click the **Create** button.
 
-#### Fetch token
+#### Fetch OAuth2 token
 
-1. In the **Exposed APIs** section, click on the API you exposed.
-2. In the **Security** section, click the **Fetch token** button.
-3. Select the whole text and copy it to the clipboard, or click the **Copy to clipboard** button. Make sure that the word `Bearer` is also copied.
-4. The token is later referred to as **\{token\}**.
+1. On the Namespace main page, click **Deploy new resource** button, select the `oauth2client.yaml` file from the `service` directory in this example, and click **Upload**.
+2. Fetch the access token with required scopes. The access token in the response is later referred to as **\{oauth2-token\}**. Run:
 
-#### Test the APIs with authentication
+    ```bash
+    curl https://oauth2.{domain}/oauth2/token -H "Authorization: Basic ZXhhbXBsZS1pZDpleGFtcGxlLXNlY3JldA==" -F "grant_type=client_credentials" -F "scope=read write"
+    ```
+
+#### Test the APIs with OAuth2 authentication
 
 ```bash
 # To perform a test without the token, use the following command:
-curl -ik https://{hostname}.kyma.local
-# > 401 Origin authentication failed.
+curl https://{hostname}.{domain}/orders
+# > {"error":{"code":401,"status":"Unauthorized","request":"1915853b-9780-4751-b26d-903a179e2941","message":"The request could not be authorized"}}
 
 # To perform a test with the token, use the following command:
-curl -ik https://{hostname}.kyma.local -H 'Authorization: {token}'
-# > 200 Hello world
+curl -ik https://{hostname}.{domain}/orders -H 'Authorization: Bearer {oauth2-token}'
+# > 200 []
 ```
+ 
+>**NOTE:** If you are using the Kyma deployed locally, add the `{hostname}.{domain}` to your hosts file.
 
 ### Manual exposure using kubectl
 
-There are additional prerequisites to exposing a service or a function manually using kubectl:
+There are additional prerequisites to exposing a service manually using kubectl:
 
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version 1.10.0
-- A token fetched from the Console UI which is later referred to as **\{token\}**. For more details, see the **NOTE** in the **Fetch token** section.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) in version specified in the [Kyma documentation](https://kyma-project.io/docs/#installation-install-kyma-locally). It needs to be configured to point to your Kyma cluster. For more info, see [this document](https://kyma-project.io/docs/components/security/#details-iam-kubeconfig-service-get-the-kubeconfig-file-and-configure-the-cli).
```

Comment:
```suggestion
- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) in the version specified in the [Kyma documentation](https://kyma-project.io/docs/#installation-install-kyma-locally). It must be configured to point to your Kyma cluster. For more information, see the document about [getting the kubeconfig file](https://kyma-project.io/docs/components/security/#details-iam-kubeconfig-service-get-the-kubeconfig-file-and-configure-the-cli).
```

Reply:
We're changing our approach to links in docs to make them more descriptive. Here is the [issue](https://github.com/kyma-project/kyma/issues/8342).

Reply:
I wasn't aware of this change, thanks!